odd-platform-ui/src/components/DataEntityDetails/QualityTestRunsHistory/TestRunsHistory.tsx

```diff
@@ -27,22 +32,50 @@ const TestRunsHistory: React.FC<QualityTestHistoryProps> = ({
   dataQATestRunsList,
   isTestRunsListFetching,
   fetchDataSetQualityTestRuns,
+  pageInfo,
 }) => {
-  React.useEffect(() => {
-    fetchDataSetQualityTestRuns({ dataqatestId: dataQATestId });
-  }, [fetchDataSetQualityTestRuns, dataQATestId]);
+  const pageSize = 100;
 
   const [alertStatus, setAlertStatus] = React.useState<
-    DataQualityTestRunStatusEnum | 'All'
+    DataQualityTestRunStatus | 'All'
   >('All');
 
-  const filterDataQATestRunsByStatus = (
-    dataQATestRun: DataQualityTestRun
-  ) => {
-    if (alertStatus === 'All') return true;
-    return dataQATestRun.status === alertStatus;
+  const statusForFetchingRuns =
+    alertStatus === 'All' ? undefined : alertStatus;
+
+  React.useEffect(() => {
```

Comment:
I'd suggest that we combine the following two blocks and get rid of `statusForFetchingRuns`. For example:
```
React.useEffect(() => {
   fetchPage(1);
}, [fetchDataSetQualityTestRuns, dataQATestId, alertStatus]);

const fetchPage = (page?: number) => {
    fetchDataSetQualityTestRuns({
      dataqatestId: dataQATestId,
      page: page || pageInfo.page + 1,
      size: pageSize,
      status: alertStatus === 'All' ? undefined : alertStatus,
    });
  };
```